theme/static/js/hs_file_browser.js

```diff
@@ -885,6 +886,36 @@ function bindFileBrowserItemEvents() {
                                 break;
                             }
                         }
+                        if (extensionApp) {
+                            // check for appkey match
+                            if ($(this).attr("data-tool-appkey")) {
+                                toolAppKey = $(this).attr("data-tool-appkey");
+                            }
+                            if (aggrAppKey && toolAppKey) {
+                                // selected file is part of an aggregation and the aggregation has an appkey
+                                // tool appkey and aggregation appkey must match for the tool to be available
+                                if (aggrAppKey !== toolAppKey) {
+                                    extensionApp = false;
+                                }
+                            }
+                            else if (toolAppKey) {
+                                // tool has appkey but there is no aggregation appkey for the selected file
+                                // tool is not applicable/viewable for the selected file
+                                extensionApp = false;
+                            }
+                            else if (fileAggType) {
+                                // check for aggregation type match
+                                if ($(this).attr("data-agg-types")) {
```

Comment:
Could use `.data()` but I think your use of `.attr()` is appropriate since we store directly on the element
https://api.jquery.com/data/